### PR TITLE
Update ESLint

### PR DESCRIPTION
**Pull request by ooflorent**
Also bump side dependencies

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,19 +1,18 @@
 {
   "extends": "eslint-config-rackt",
+  "parser": "babel-eslint",
```
Comment (ooflorent) [inline .eslintrc]: Explicitly use Babel for parsing

File: .eslintrc [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
-    "react/jsx-uses-react": 1,
+    "react/jsx-uses-react": 2,
+    "react/jsx-uses-vars": 2,
     "react/jsx-no-undef": 2,
     "react/wrap-multilines": 2
   },
   "plugins": [
     "react"
-  ],
-  "globals": {
```
Comment (ooflorent) [inline .eslintrc]: This global variable is not used

File: src/reduxForm.js [hunk 3]
```diff
@@ -255,6 +255,7 @@ const createReduxForm =
 
           render() {
             // remove some redux-form config-only props
+            /* eslint-disable no-unused-vars */
```
Comment (ooflorent) [inline src/reduxForm.js]: Here we use destructuring to compute `passableProps`. We need this to mute ESLint because we create a lot of unused bindings.

Comment (erikras) [inline src/reduxForm.js]: That's dumb on eslint's part.

Comment (ooflorent) [inline src/reduxForm.js]: This won't be addressed until _Rest/Spread Properties_ would progress to higher stage.
See https://github.com/eslint/eslint/issues/4880#issuecomment-169860909.

**Pull request merged**